### PR TITLE
Generating function component by default

### LINKED ISSUE #1067: Suggestion to add functional option to component generator
Thank you for this wonderful library, it's helped me set up React in Rails when I'm new to Rails. It's especially nice to have `rails g react:component` for quickly creating new components. I'm using a lot of functional components in my project, though, so it's not as useful.

What about adding an option to make it a functional component? Something like `rails g react:component --func` or `rails g react:component --function`.

This would create:
```jsx
import React from "react"
import PropTypes from "prop-types"
function SomeComponent() {
  return (
    <React.Fragment>
      Example: {this.props.example}
    </React.Fragment>
  );
}

SomeComponent.propTypes = {
  example: PropTypes.text
};
export default SomeComponent

```

### PR DESCRIPTION
**Pull request by ahangarha**
### Summary

This PR changes the functionality of component generation by creating function components.

### Pull Request checklist

- [x] Add/update test to cover these changes
- [x] Update documentation
- [x] Update CHANGELOG file

### Other Information

Closes #1067

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ _Please add entries here for your pull requests that are not yet released._
 - Include turbolinks in dev and update webdrivers #1174
 - Add support for multiple `require.context` with addition of `useContexts` #1144
 - Update dependencies
+- Add `--function` and `--func` options for generating a React function component #1271
```
Comment (justin808) [inline CHANGELOG.md:20]: Let's make this the default.

We can target this for 3.0.

Comment (ahangarha) [reply CHANGELOG.md]: Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,9 @@ _Please add entries here for your pull requests that are not yet released._
 #### Bug Fixes
 - Fix ReactDomClient initialization error during SSR. #1278
 
+#### Changed
+- The `react:component` generator now generates a function component by default #1271
+
```
Comment (Judahmeek) [inline CHANGELOG.md:26]: Shouldn't this PR go under "unreleased" rather then under 2.7.1, which was released last month?

**Review by Judahmeek — CHANGES_REQUESTED** [review]
One question on Changelog entry.

**Pull request merged**